tiny_dnn/core/kernels/global_avepool_op_avx.h

```diff
@@ -0,0 +1,120 @@
+/*
+    Copyright (c) 2017, Taiga Nomi
+    All rights reserved.
+
+    Use of this source code is governed by a BSD-style license that can be found
+    in the LICENSE file.
+*/
+#pragma once
+
+namespace tiny_dnn {
+namespace kernels {
+
+#ifdef CNN_USE_AVX
+
+inline void avx_global_avepool_forward_kernel(
+  const tensor_t &in_data,
+  tensor_t &out_data,
+  const core::global_avepool_params &params,
+  const bool layer_parallelize) {
+  const size_t pool_area            = params.in.width_ * params.in.height_;
+  const float_t pool_area_inv       = 1.0f / pool_area;
```

Comment:
You don't need to use `float_t` in here. Just `float` is OK. And don't forget to cast `pool_area` to `float` with `static_cast`, otherwise I think compiler gives a warning.

I'd use `_mm_set_ss` to hold `pool_area` value in `__m128` typed variable and use `_mm_rcp_ss` to compute a reciprocal. In that way, you can multiply it with the result of `hsum256_ps` with `_mm_mul_ss`.